tasks/install_powershell.ps1

```diff
@@ -107,18 +107,25 @@ function DownloadPuppet {
   Write-Verbose "Downloading the Puppet Agent for Puppet Enterprise on $env:COMPUTERNAME..."
 
   $webclient = New-Object system.net.webclient
-
-  try {
-    $webclient.DownloadFile($msi_source,$msi_dest)
-  }
-  catch [System.Net.WebException] {
-    # If we can't find the msi, then we may not be configured correctly
-    if($_.Exception.Response.StatusCode -eq [system.net.httpstatuscode]::NotFound) {
-        Throw "Failed to download the Puppet Agent installer: $msi_source"
+  
+  $a = 0
```

Comment:
nit: Something like `$attempt_number` instead of `$a`? It fits better with the rest of the code and would be easier to read the code.

Reply:
good observation!